src/components/AutoComplete/AsyncAutoComplete.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { AsyncTypeahead } from 'react-bootstrap-typeahead';
+import PropTypes from 'prop-types';
+
+class AsyncAutoComplete extends React.Component {
+  state = {
+    options: this.props.options,
+    isLoading: this.props.isLoading
+  };
+
+  onSearchStart = () => this.setState({ isLoading: true });
+
+  onSearchEnd = options => this.setState({ options, isLoading: false });
+
+  handleSearch = () => {
+    this.onSearchStart();
+    Promise.resolve(this.props.onSearch()).then(options =>
```

Comment:
Thanks for making this change.  AsyncTypeahead passes a query (the value of the input) to `onSearch`.  We should probably pass that into this.props.onSearch as well.  That way a consumer can do server-side filtering based of this as well. similar to the example here: https://github.com/ericgio/react-bootstrap-typeahead/blob/master/docs/API.md#asynctypeahead we dont need to do it in storybook, but we just was to offer it.